src/decoder.mli

```diff
@@ -2,16 +2,18 @@ open S
 
 type t
 
-val create : int -> t
-(** [create capacity] intializes a decoder with a dynamic table with maximal
-    size [capacity]. This size is an approximation of the memory usage in
-    bytes. *)
-
-val set_capacity : t -> int -> unit
-(** [set_capacity decoder capacity] updates the capacity of the dynamic table. *)
+val create : ?max_size_limit:int -> unit -> t
+(** [create ~max_size_limit ()] intializes a decoder with a dynamic table with
+    maximum size [max_size_limit]. The default is 4096. Each different
+    should be communicated to the encoder by the protocol. *)
 
 val header : t -> header Angstrom.t
 (** A parser for one header *)
 
 val headers : t -> header list Angstrom.t
-(** A parser for headers *)
+(** A parser for one header block *)
+
+val change_table_size_limit : t -> int -> unit
```

Comment:
I'm not sure if this is actually needed to be on the public interface. My understanding might be wrong (and I'll re-read the RFC in a sec) but I think a prefix of `0x20` signals a dynamic table size update at the beginning of a header block, and therefore an API is not needed?

Reply:
There is a difference between the maximum size chosen by the decoder (which I call `max_size_limit`), and the maximum size used by the encoder (`max_size`, which has to be lower than `max_size_limit`, but doesn't have to be equal).